src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -40,6 +46,21 @@ public DocumentDBConfiguration()
         /// </summary>
         public string ConnectionString { get; set; }
 
+        /// <summary>
+        /// Gets or sets the DocumentDB connection string for the monitored service.
+        /// </summary>
+        public string TriggerConnectionString { get; set; }
+
+        /// <summary>
+        /// Optional. Gets or sets the DocumentDB connection string for the leases service.
+        /// </summary>
+        public string LeasesConnectionString { get; set; }
```

Comment:
I'm curious what @MikeStall thinks, but I think we can probably get rid of the `Trigger` and `Lease` connection string properties here and have the most common scenario work like:

- You use the config.ConnectionString value (which falls back to the environment variable) as the fallback for any missing connection string.
- If you want to override any of these in a specific function, you use the `triggerAttribute.ConnectionString` or `triggerAttribute.LeaseConnectionString` to do so.

That keeps the concepts a little cleaner in the configuration and still allows you to achieve everything you need to. Yes, there's still a chance that you accidentally wire up a loop, but that's true of a lot of our bindings that have triggers.

Plus, if we leave them off for now, we can always add the properties in the future if we decide we need them (while taking them away is harder).